### PR TITLE
GT4SD inference client

### PR DESCRIPTION
**Pull request by drugilsberg**
Initial version of the inference client.

### CODE REVIEW

**Comment by drugilsberg** [comment]
Closes #5

**Comment by drugilsberg** [comment]
Some commands to test it:

Help:

```console
gt4sd-inference --help
```

From the README.md.

Get info:

```console
gt4sd-inference --algorithm_name PaccMannRL --algorithm_application PaccMannRLProteinBasedGenerator --print_info
```

Run PaccMannRL:

```console
gt4sd-inference --algorithm_name PaccMannRL --algorithm_application PaccMannRLProteinBasedGenerator --target MVLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTT --number_of_samples 10
```

Run PolymerBlocks

```console
gt4sd-inference --algorithm_name PolymerBlocks --algorithm_application PolymerBlocksGenerator --number_of_samples 10
```

Pass incomplete arguments.

Match all algorithms fitting and print info on available ones:

```console
gt4sd-inference --algorithm_name PaccMannRL
```

List all:

```console
gt4sd-inference
```

Impossible request:

```console
gt4sd-inference --algorithm_name PaccMannRL --domain nlp
```

**Review by christofid — APPROVED** [review]
Looks good to me!

File: README.md [hunk 1]
```diff
@@ -100,6 +100,97 @@ items = list(algorithm.sample(10))
 print(items)
 ```
 
+### Running inference pipelines via the CLI command
+
+GT4SD can run inference pipelines based on the `gt4sd-infrence` CLI command.
```
Comment (drugilsberg) [inline README.md:105]:
```suggestion
GT4SD can run inference pipelines based on the `gt4sd-inference` CLI command.
```

**Pull request merged**